lib/openstax/rescue_from.rb

```diff
@@ -0,0 +1,136 @@
+require 'openstax/rescue_from/configuration'
+
+module OpenStax
+  module RescueFrom
+    class << self
+      def perform_rescue(exception:, listener: MuteListener.new)
+        proxy = ExceptionProxy.new(exception)
+
+        listener.before_openstax_exception_rescue(proxy)
+        register_exception(exception.class)
+        log_system_error(proxy)
+        send_notifying_exceptions(proxy, listener)
+        finish_exception_rescue(proxy, listener)
+      end
+
+      def register_exception(exception, options = {})
+        unless registered_exceptions.include?(exception.name)
```

Comment:
I would prefer for any call to `register_exception` to override the options provided in a previous call.  That way we can have a bunch of defaults inside the gem, at least for all of the standard Rails exceptions, but their behavior can be overriden on an app-by-app basis if need be.  We don't want to register ActiveRecord::NotFound in each app, but want the flexibility for an app to override that exception's default behavior if it wants.

So to do this, the `unless` would have to come out of here and you'd have to find the options in each of lines 23-26 and back them out and reset them.  

Instead of doing that, I think it would be easier to have an ExceptionOptions object for each exception name.  Then if an exception is registered, you can just pull it out of a list and put in the new options (instead of having to modify four different sets / arrays.